### PR TITLE
Add articles to item from item page

### PR DESCRIPTION
**Pull request by caiosba**
## Description

See Figma designs in referenced Jira ticket for details. But the main ideas here are:

- Delete the left sidebar from the item page (the one with claim and fact-check)
- Add a new right sidebar called "Articles"
- In that sidebar, for items with no articles (so, no explainers and no fact-check), show a button to create new article and list articles to be applied to item
- Clicking on an article (explainer or fact-check) adds it to the current item
- The articles in that right sidebar have a hover style
- When the item has an article, the right sidebar shows only the "New article" button - this case is going to be handled by another ticket

Reference: CV2-4626

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Unit tests added for new card component.

## Things to pay attention to during code review

May require a branch in Check API with the same name as this one.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [x] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- x ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

**Comment by caiosba** [comment]
Update for reviewers: the backend for this feature is now in the epic branch (`epic/CV2-4441-articles`).

**Review by brianfleming — APPROVED** [review]
As discussed with @caiosba this looks good on my end with the change to the default tab, then wait on a tech reviewer to add any comments

**Review by danielevalverde — COMMENTED** [review]
@caiosba, when I tried to create a "new claim & fact check" in an import fact-check item type, I got the error message: 
`Deferred.js:67 Uncaught TypeError: Cannot read properties of null (reading 'claim_description')   at handleSaveFactCheck (ClaimFactCheckForm.js:147:1)`

Steps to reproduce: 
- go to an imported fact check item page
- click to add a new claim and fact check
- fill in the fields and click to create the comment

Captura de tela_2024-06-24_11-17-32

**Review by danielevalverde — COMMENTED** [review]
@caiosba, While testing further, I noticed that the issue isn't limited to fact check item types. I'm unable to create a new article from either the item page or the list page.

Upon debugging, I found that the response we pass to the handleSaveFactCheck function is returning project_media as null.
Captura de tela_2024-06-24_12-18-53

We rely on the project_media data to send variables in the mutation: 
https://github.com/meedan/check-web/blob/epic/CV2-4441-articles/src/app/components/article/ClaimFactCheckForm.js#L141-L168

Could this issue be related to the recent back-end changes? https://github.com/meedan/check-api/commit/e0c81281c9b4dfc208bcf1bf0cd0016cb0d7d136

and question to @brianfleming:
- Should we be able to add a "Claim & Fact Check" to an item that has the type "imported fact check"? Sorry, but It seems a bit odd to me to add a fact check to a fact check, so I wanted to clarify :sweat_smile: 
- Should we be able to add an article to an item that is in the trash?

**Comment by caiosba** [comment]
Thanks @danielevalverde ! Yes, creating articles is out of the scope of this ticket. There is glue work that needs to be done between this work (CV2-4626) and @sarahkeh 's work on the forms (CV2-4447 and CV2-4498), but the forms need to the ready for that. Not only for this PR, but also for editing articles from the articles list. I'm keeping in the epic ticket (CV2-4441) a list of glue work that needs to be done... if you see anything missing, please add there! :)

For the scope of this ticket, you're right, I need to hide the "create" button for some cases, probably. I'll wait for @brianfleming 's input.

**Comment by brianfleming** [comment]
> Thanks @danielevalverde ! Yes, creating articles is out of the scope of this ticket. There is glue work that needs to be done between this work ([CV2-4626](https://meedan.atlassian.net/browse/CV2-4626)) and @sarahkeh 's work on the forms ([CV2-4447](https://meedan.atlassian.net/browse/CV2-4447) and [CV2-4498](https://meedan.atlassian.net/browse/CV2-4498)), but the forms need to the ready for that. Not only for this PR, but also for editing articles from the articles list. I'm keeping in the epic ticket ([CV2-4441](https://meedan.atlassian.net/browse/CV2-4441)) a list of glue work that needs to be done... if you see anything missing, please add there! :)
> 
> For the scope of this ticket, you're right, I need to hide the "create" button for some cases, probably. I'll wait for @brianfleming 's input.

Ok, discussed on huddle:
- Trash you will be able to do whatever you want, no need to restrict
- Imported fact-checks going forward will not create a blank Media (@caiosba will create ticket to address with backend changes and API deprecation warnings, etc)
- Existing Imported fact-checks that already created blank media...we will disable the necessary buttons on the media item page with a message to users about it being imported

**Comment by caiosba** [comment]
@brianfleming @danielevalverde , I pushed the fixes based on the comments here and the things we talked about:

- Button disabled with tooltip on legacy imported item page:
Captura de tela de 2024-06-24 19-14-05
- The "Articles" tab is the default now

**Review by sarahkeh — COMMENTED** [review]
Reusing the NewArticleButton is smart.  With the changes you made to the ArticleForm, did you check that the create button works in the list view as well?

**Comment by amoedoamorim** [comment]
Will review tomorrow first thing in the morning

File: localization/react-intl/src/app/components/article/ArticlesSidebar.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[
+  {
+    "id": "articlesSidebar.success",
+    "description": "Banner displayed after an article is successfully added to an item.",
+    "defaultMessage": "Article added successfully!"
+  },
+  {
+    "id": "articlesSidebar.noArticlesAddedToItem",
+    "description": "Message displayed on articles sidebar when an item has no articles.",
+    "defaultMessage": "No articles are being delivered to Tipline users who send in request with that match this Media."
```
Comment (amoedoamorim) [inline localization/react-intl/src/app/components/article/ArticlesSidebar.json:10]: I found this sentence a bit weird. How about `...who send requests that match this Media."

File: src/app/components/media/MediaComponentRightPanel.js [hunk 2]
```diff
@@ -89,6 +104,7 @@ const MediaComponentRightPanel = ({
       { showTab === 'suggestedMedia' ? <MediaSuggestions dbid={projectMedia.dbid} teamDbid={projectMedia.team?.dbid} superAdminMask={superAdminMask} /> : null }
       { showTab === 'metadata' ? <MediaTasks media={projectMedia} fieldset="metadata" /> : null }
       { showTab === 'source' ? <MediaSource projectMedia={projectMedia} /> : null }
+      { showTab === 'articles' ? <ArticlesSidebar teamSlug={projectMedia.team.slug} projectMediaDbid={projectMedia.dbid} /> : null }
```
Comment (amoedoamorim) [inline src/app/components/media/MediaComponentRightPanel.js:107]: Maybe call this component `MediaArticles` for consistency?

**Review by amoedoamorim — COMMENTED** [review]
Adding a couple of comments but still working on reviewing

File: src/app/components/article/ArticleForm.js [hunk 3]
```diff
@@ -35,7 +36,7 @@ const ArticleForm = ({
 
   const [claimDescription, setClaimDescription] = React.useState(articleType === 'fact check' && projectMedia?.suggested_main_item ? projectMedia.suggested_main_item.claim_description : '');
```
Comment (amoedoamorim) [inline src/app/components/article/ArticleForm.js:37]: `fact-check` to match propType

File: src/app/components/media/MediaComponentRightPanel.js (hunk 2 above)
Comment (caiosba) [reply src/app/components/media/MediaComponentRightPanel.js:107]: Sure, I'll rename!

File: localization/react-intl/src/app/components/article/ArticlesSidebar.json (hunk 1 above)
Comment (caiosba) [reply localization/react-intl/src/app/components/article/ArticlesSidebar.json:10]: Defer to @brianfleming - I just copied from the design.

File: src/app/components/article/ArticlesSidebar.js [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+  return (
+    <div id="articles-sidebar" className={styles.articlesSidebar}>
+      <div className={styles.articlesSidebarTopBar}>
+        {/* FIXME: Make sure the form can receive the right reference for the current item */}
+        <NewArticleButton team={team} buttonMainProps={{ size: 'small', theme: 'text' }} disabled={projectMedia.type === 'Blank'} />
+      </div>
+      { !hasArticle && (
+        <>
+          <div className={cx('typography-body1', styles.articlesSidebarNoArticle)}>
+            <DescriptionIcon style={{ fontSize: '32px' }} />
```
Comment (amoedoamorim) [inline src/app/components/article/ArticlesSidebar.js:125]: Maybe we're missing a matching `iconSize...` css variable @brianfleming

File: src/app/components/article/ArticlesSidebar.js [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+  }).isRequired,
+  onUpdate: PropTypes.func.isRequired,
+};
+
+const ArticlesSidebar = ({ teamSlug, projectMediaDbid }) => {
+  const [updateCount, setUpdateCount] = React.useState(0);
+
+  const handleUpdate = () => {
+    setUpdateCount(updateCount + 1);
+  };
```
Comment (amoedoamorim) [inline src/app/components/article/ArticlesSidebar.js:169]: question (non-blocking): why is this necessary?

File: localization/react-intl/src/app/components/article/ArticlesSidebar.json (hunk 1 above)
Comment (brianfleming) [reply localization/react-intl/src/app/components/article/ArticlesSidebar.json:10]: Oh yeah, that is a typo for sure. Thats what I get for not proofreading designs. @amoedoamorim is correct, lets go with:

`No articles are being delivered to Tipline users who send requests that match this Media.`

File: src/app/components/article/ArticlesSidebar.js (hunk 4 above)
Comment (brianfleming) [reply src/app/components/article/ArticlesSidebar.js:125]: Good catch @amoedoamorim 
@caiosba you can change this and put it in the css file and use the var:

`font-size: var(--font-size-h4);`

**Review by amoedoamorim — CHANGES_REQUESTED** [review]
Done reviewing. Looks great, just needs a simple copy fix and typo fix in `fact check`.

File: localization/react-intl/src/app/components/article/ArticlesSidebar.json (hunk 1 above)
Comment (caiosba) [reply localization/react-intl/src/app/components/article/ArticlesSidebar.json:10]: Fixed!

File: src/app/components/article/ArticleForm.js (hunk 3 above)
Comment (caiosba) [reply src/app/components/article/ArticleForm.js:37]: I thought about this too, but I didn't want to change too much in the forms in the scope of this ticket since @sarahkeh is still working on them.

Comment (sarahkeh) [reply src/app/components/article/ArticleForm.js:37]: This is something I also did in my branch, so we're doubly covered

File: src/app/components/article/ArticlesSidebar.js (hunk 4 above)
Comment (caiosba) [reply src/app/components/article/ArticlesSidebar.js:125]: @brianfleming `--font-size-h4` is 34px while the icon on the design was 32px - is it OK to keep it 34px?

Comment (brianfleming) [reply src/app/components/article/ArticlesSidebar.js:125]: yes! I checked it in the Figma design file and it works fine

File: src/app/components/article/ArticleForm.js (hunk 3 above)
Comment (caiosba) [reply src/app/components/article/ArticleForm.js:37]: Awesome, thanks @sarahkeh !

File: src/app/components/article/ArticlesSidebar.js (hunk 4 above)
Comment (caiosba) [reply src/app/components/article/ArticlesSidebar.js:125]: Thanks, done!

File: src/app/components/article/ArticleForm.js (hunk 3 above)
Comment (amoedoamorim) [reply src/app/components/article/ArticleForm.js:37]: Awesome! :bananadance:

File: src/app/components/article/ArticlesSidebar.js (hunk 5 above)
Comment (caiosba) [reply src/app/components/article/ArticlesSidebar.js]: Shouldn't be. But for some reason, for now, the Relay store updates were not enough. I added a `FIXME` to it to see if we can remove it after the rest of the feature is implemented. Once we have the components to list the item's articles, with the proper fragment, I hope we can delete this "hack".

**Comment by caiosba** [comment]
Thanks @amoedoamorim , please re-review.

File: localization/react-intl/src/app/components/article/ArticlesSidebar.json (hunk 1 above)
Comment (amoedoamorim) [reply localization/react-intl/src/app/components/article/ArticlesSidebar.json]: Noice!

File: src/app/components/media/MediaComponentRightPanel.js (hunk 2 above)
Comment (amoedoamorim) [reply src/app/components/media/MediaComponentRightPanel.js]: This is beauty in component naming form! :star_struck:

File: src/app/components/article/ArticlesSidebar.js (hunk 5 above)
Comment (amoedoamorim) [reply src/app/components/article/ArticlesSidebar.js]: Awesome, thanks for clarifying!

**Review by amoedoamorim — APPROVED** [review]
:lgtm:
